### ISSUE TITLE
--eexit does not work as specified

### ISSUE THREAD

**Issue by Sellitus** [nanopool/ewbf-miner#9]
When using the --eexit command with any argument following it, it does not close the miner upon disconnection. This makes auto-switching pools impossible to use EWBF's miner on.

**Comment by voltcode** [comment]
I second that. When it hangs , I can press enter for it to reconnect. Without pressing enter, ewbf just hangs. Doesn't matter if I set --eexit or not

**Comment by hubdr** [comment]
Same here. It does NOT EXIT after server a disconnect  even though i have --exit 3 
Keeps trying to reconnect  with "attempt to restore connection"

**Comment by kkniffin** [comment]
I am also having the same issue. Has anyone found a good workaround in the mean time?

**Comment by TestusRus** [comment]
Please add possibility to exit when any error occurs during mining process.

**Comment by vla7** [comment]
Same problem on Linux version.

**Comment by pronobis** [comment]
Yes, this is quite critical. Unable to use the miner for autoswitching.

**Comment by ark02** [comment]
Has anyone found a workaround for this issue yet? 

If not, do you know of any zcash miners with working exit functionality?

**Comment by pronobis** [comment]
I just run this bash script instead:

```
(PIDFILE=$(mktemp /tmp/foo.XXXXXX) && trap "rm $PIDFILE" 0 \
         && { (unbuffer <YOUR_EWBF_CMD_HERE>) \
                  1> >(tee >(grep -q ERROR && kill $(cat $PIDFILE)) >&1) \
              & PID=$! && echo $PID >$PIDFILE ; wait $PID || true; })
```

It detects the word ERROR in the output and kills EWBF.

**Comment by totembe** [comment]
I run EWBF miner on both linux and windows. Multi-algo switching fails due to this problem.  Needs to be fixed.

**Comment by ark02** [comment]
@pronobis thank you for this gem. Fixed all my issues.

**Comment by ark02** [comment]
@pronobis how would one pass multiple catches to trigger the kill command? Can you nest that code within itself? 

Probably better question: Can you point to any solid references that can help a shell newbie understand what you did there?

**Comment by pronobis** [comment]
If I understand you correctly, you can just modify the call to grep to look for multiple patterns. Check this out: https://unix.stackexchange.com/questions/37313/how-do-i-grep-for-multiple-patterns

**Comment by DeviousPenguin** [comment]
hubdr: try setting --eexit to 2

from the Help.txt file:
"--eexit             Exit in case of error. Value 1 exit if miner cannot restart workers.
                    Value 2 if lost connection with the pool. 3 both cases."

**Comment by zaff01** [comment]
any way to do this with windows ?
